testing/tests/DevExpress.ui.widgets.dataGrid/editing.tests.js

```diff
@@ -9012,6 +9012,333 @@ QUnit.module('Editing with real dataController', {
+
+            // assert
+            assert.ok($(this.getRowElement(0)).hasClass('dx-edit-row'), 'row is edited');
+            assert.equal(onSaving.callCount, 1, 'onSaving');
+            assert.equal(onSaved.callCount, 0, 'onSaved');
+            assert.equal(onEditCanceling.callCount, 0, 'onEditCanceling');
+            assert.equal(onEditCanceled.callCount, 0, 'onEditCanceled');
+
+            // act
+            this.clock.tick(500);
+            this.cancelEditData();
+
+            // assert
+            assert.ok($(this.getRowElement(0)).hasClass('dx-edit-row'), 'row is edited');
+            assert.equal(onSaving.callCount, 1, 'onSaving');
+            assert.equal(onSaved.callCount, 0, 'onSaved');
+            assert.equal(onEditCanceling.callCount, 1, 'onEditCanceling');
+            assert.equal(onEditCanceled.callCount, 0, 'onEditCanceled');
+
+            // act
+            this.clock.tick(500);
+
+            // assert
+            assert.notOk($(this.getRowElement(0)).hasClass('dx-edit-row'), 'row is not edited');
+            assert.equal(onSaving.callCount, 1, 'onSaving');
+            assert.equal(onSaved.callCount, 1, 'onSaved');
+            assert.equal(onEditCanceling.callCount, 1, 'onEditCanceling');
+            assert.equal(onEditCanceled.callCount, 0, 'onEditCanceled');
+
+            // act
+            this.cancelEditData();
+
+            // assert
+            assert.equal(onEditCanceling.callCount, 2, 'onEditCanceling');
+            assert.equal(onEditCanceled.callCount, 1, 'onEditCanceled');
+        });
+
+        QUnit.test('Promise in onSaving with cancel', function(assert) {
+            // arrange
+            const rowsView = this.rowsView;
+            const $testElement = $('#container');
+            const onSaving = sinon.spy(e => {
+                e.promise = new Deferred();
+                setTimeout(() => {
+                    e.cancel = true;
+                    e.promise.resolve();
+                }, 1000);
+            });
+            const onSaved = sinon.spy();
+
+            this.options.editing = {
+                allowUpdating: true
+            };
+            this.options.onSaving = onSaving;
+            this.options.onSaved = onSaved;
+            this.editingController.optionChanged({ name: 'onSaving' });
+            this.editingController.optionChanged({ name: 'onSaved' });
+            rowsView.render($testElement);
+
+            // act
+            this.editRow(0);
+            this.cellValue(0, 0, 'new value');
+            this.saveEditData();
+
+            // assert
+            assert.ok($(this.getRowElement(0)).hasClass('dx-edit-row'), 'row is edited');
+            assert.equal(onSaving.callCount, 1, 'onSaving');
+            assert.equal(onSaved.callCount, 0, 'onSaved');
+
+            // act
+            this.clock.tick(1000);
+
+            // assert
+            assert.ok($(this.getRowElement(0)).hasClass('dx-edit-row'), 'row is edited');
```

Comment:
Asserts are not changed after tick(1000). Why? I think we should check that load panel is hidden after this tick.